cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -4556,22 +4566,22 @@ packageHashConfigInputs shared@ElaboratedSharedConfig{..} pkg =
     , pkgHashPlatform = pkgConfigPlatform
     , pkgHashFlagAssignment = elabFlagAssignment
     , pkgHashConfigureScriptArgs = elabConfigureScriptArgs
-    , pkgHashVanillaLib = elabVanillaLib
-    , pkgHashSharedLib = elabSharedLib
-    , pkgHashDynExe = elabDynExe
-    , pkgHashFullyStaticExe = elabFullyStaticExe
-    , pkgHashGHCiLib = elabGHCiLib
-    , pkgHashProfLib = elabProfLib
-    , pkgHashProfExe = elabProfExe
-    , pkgHashProfLibDetail = elabProfLibDetail
-    , pkgHashProfExeDetail = elabProfExeDetail
-    , pkgHashCoverage = elabCoverage
-    , pkgHashOptimization = elabOptimization
-    , pkgHashSplitSections = elabSplitSections
-    , pkgHashSplitObjs = elabSplitObjs
-    , pkgHashStripLibs = elabStripLibs
-    , pkgHashStripExes = elabStripExes
-    , pkgHashDebugInfo = elabDebugInfo
+    , pkgHashVanillaLib = withVanillaLib
+    , pkgHashSharedLib = withSharedLib
+    , pkgHashDynExe = withDynExe
+    , pkgHashFullyStaticExe = withFullyStaticExe
+    , pkgHashGHCiLib = withGHCiLib
+    , pkgHashProfLib = withProfLib
+    , pkgHashProfExe = withProfExe
+    , pkgHashProfLibDetail = withProfLibDetail
+    , pkgHashProfExeDetail = withProfExeDetail
+    , pkgHashCoverage = exeCoverage
+    , pkgHashOptimization = withOptimization
+    , pkgHashSplitSections = splitSections
+    , pkgHashSplitObjs = splitObjs
+    , pkgHashStripLibs = stripLibs
+    , pkgHashStripExes = stripExes
+    , pkgHashDebugInfo = withDebugInfo
```

Comment:
Maybe we can group these too? But that type belongs to cabal-install this time because it's cabal-install that decides what needs to be recompiled or not.

Reply:
I'll leave that as-is for now.